src/components/multi_select_state.js

```diff
@@ -32,20 +33,69 @@ const withMultiSelectState = WrappedComponent =>
       }
     }
 
-    selectItem(id) {
-      const { items } = this.props;
-      const { selectedItems } = this.state;
-      if (!selectedItems.find(item => item.id === id)) {
-        const newSelectedItems = items.filter(
-          item =>
-            item.id === id ||
-            selectedItems.find(selectedItem => item.id === selectedItem.id)
-        );
-        this.setState({ selectedItems: newSelectedItems }, this.handleChange);
-      } else {
-        this.unselectItem(id);
+      handleMultiSelection(id) {
+        const { items } = this.props;
+          const { selectedItems, firstItemShiftSelected, filteredItems } = this.state;
+          let newSelectedItems;
+          if(firstItemShiftSelected == null) {
+              this.setState({ firstItemShiftSelected: id });
+              newSelectedItems = items.filter(
+                  item =>
+                      item.id === id ||
+                      selectedItems.find(selectedItem => item.id === selectedItem.id)
+              );
+          } else {
+              if (id > firstItemShiftSelected) {
```

Comment:
Your assumption here is that id is a number and is ordered.
I don't think this is the case...
In my opinion we should use their index in the filteredItems.

Reply:
All this logic should be streamlined.
I think the moment you'll move to index based search you can simply go from min to max instead of this if.